src/components/Login.jsx

```diff
@@ -1,188 +1,57 @@
-			setLoginErrorText("");
-		} else {
-			setError(false);
-			setPasswordErrorText("");
-			setLoginErrorText("");
-		}
-
-		setPassword(pass);
-
-		if (event.charCode === 13) {
-			loginWrapper(event);
-		}
-	};
-
-	const loginWrapper = async () => {
-		await login(username, password)(dispatch);
-	};
-
-	const handleKeyPressed = (event) => {
-		if (event.charCode === 13) {
-			loginWrapper();
-		}
-	};
-
-	// if already login, redirect to homepage
-	if (Authorization !== undefined && Authorization !== "" && Authorization !== null) {
-		browserHistory.push("/");
-		return null;
-	}
-	// else render login page
 	return (
-		<div>
-			<div className="center" style={{ display: "block", margin: "auto", width: "500px", paddingTop: "10%" }}>
-				{/*TODO: Add loading spinner here*/}
-				<Paper style={{ padding: 40 }}>
-					<Avatar style={{ margin: "auto" }}>
-						<LockOutlinedIcon />
-					</Avatar>
-					<Typography component="h1" variant="h5">
-						Sign in
-					</Typography>
-					<Divider />
-					<ImageList cols={1} cellHeight="auto">
-						<ImageListItem>
-							<p style={{ color: "red" }}>{loginErrorText} </p>
-						</ImageListItem>
-						<ImageListItem>
-							<TextField
-								variant="outlined"
-								margin="normal"
-								required
-								fullWidth
-								autoFocus
-								id="username"
-								label="Username"
-								name="username"
-								value={username}
-								onChange={changeUsername}
-							/>
-							<TextField
-								variant="outlined"
-								margin="normal"
-								required
-								fullWidth
-								id="password"
-								label="Password"
-								name="password"
-								type="password"
-								error={error}
-								helperText={passwordErrorText}
-								value={password}
-								onChange={changePassword}
-								onKeyPress={handleKeyPressed}
-							/>
-							<Link href={config.resetPwURL} className={classes.resetPW} target="_blank">
-								Forgot password?
-							</Link>
-							<Box className={classes.tos}>
-								<Typography variant="body2" style={{ display: "inline" }}>
-									By continuing, you agree to our{" "}
-								</Typography>
-								<Link href={config.tosURL} style={{ display: "inline" }} target="_blank">
-									Terms of Services
-								</Link>
-								<Typography variant="body2" style={{ display: "inline" }}>
-									&nbsp;and&nbsp;
-								</Typography>
-								<Link href={config.privacyURL} style={{ display: "inline" }} target="_blank">
-									Web Privacy Notice
-								</Link>
-							</Box>
-							<Button type="submit" fullWidth variant="contained" color="primary" onClick={loginWrapper}>
-								Sign In
-							</Button>
-							<Link href={config.signUpURL} className={classes.signUp} target="_blank">
-								Don't have an account? Sign up.
-							</Link>
-						</ImageListItem>
-					</ImageList>
-				</Paper>
-				<Version />
-			</div>
+		<div style={styles.container}>
+			<CircularProgress />
```

Comment:
The circular progress seems very convenient. Today I learned. :-)

Reply:
:D